File: src/test/test_files/java/ForLoopNotCallFuncInDeclarationRuleCodeExample.java
@SuppressWarnings("all")
class ForLoopNotCallFuncInDeclarationRuleCodeExample {
  void myMethod() {
    for (int i = myFuncInit(); i < myFuncExp(); myFuncUpdate(i)) {} // Noncompliant
    for (int j = myFuncInit(); j < 5; j++) {} // Noncompliant 
    for (String s = "abc"; s.length() > 0; s = s.substring(1)) {} // Noncompliant 
    for (int j = 0; j < 5; myFuncUpdate(j)) {} // Noncompliant 
    for (int i = myFuncInit(); i < 5; myFuncUpdate(i)) {} // Noncompliant
    for (int i = 0; i < myFuncExp(); myFuncUpdate(i)) {} // Noncompliant 
    for (int j = 0; j > 5; j--) {} 
    for (int i = 0; i < 5; ++i) {} 
  }
  int myFuncInit() {return 0;}
  int myFuncExp() {return 5;}
  int myFuncUpdate(int i) {return i++;}
}

File: src/test/test_files/php/ForLoopNotCallFuncInDeclarationRuleCodeExample.php
@SuppressWarnings("all");
<?php
    function fonction(){
        return 2;
    }
    $j=0;
    for($i = 0; $i < 10; $i++){
        $i++;
        print(fonction()); // Noncompliant
    }
    for($j = 0; $j < 10; $j++){
        $i++;
    }
    for($j = 0; $j < 10; $j++){}
    for($j = 0; fonction() < 10; $j++){ // Noncompliant
        $i++;
    }
    for($j = fonction();  $j< 10; $j++){ // Noncompliant
        $i++;
    }
    for($j = 0; $j < 10; $j+=fonction()){ // Noncompliant
        $i++;
    }
?>